hazelcast/src/main/java/com/hazelcast/map/impl/record/Records.java

```diff
@@ -29,37 +34,39 @@
     private Records() {
     }
 
-    public static void applyRecordInfo(Record record, RecordInfo replicationInfo) {
-        record.setVersion(replicationInfo.getVersion());
-        record.setHits(replicationInfo.getHits());
-        record.setTtl(replicationInfo.getTtl());
-        record.setMaxIdle(replicationInfo.getMaxIdle());
-        record.setCreationTime(replicationInfo.getCreationTime());
-        record.setLastAccessTime(replicationInfo.getLastAccessTime());
-        record.setLastUpdateTime(replicationInfo.getLastUpdateTime());
-        record.setExpirationTime(replicationInfo.getExpirationTime());
-        record.setLastStoredTime(replicationInfo.getLastStoredTime());
+    public static void writeRecord(ObjectDataOutput out, Record record, Data dataValue) throws IOException {
+        out.writeByte(record.getMatchingRecordReaderWriter().getId());
```

Comment:
The serialisation cost is the same, right? Because previously we were serialising - key, value, metadata fields by directly calling `RecordInfo.writeData` which avoids writing the class header. A boolean was also added to determine if the stream contains stats.
Now, we serialise key, value, metadata fields and an additional short ID for the writer/reader type. Am I right? Where do we save on serialization costs?

Reply:
We use internal representation of values instead of api returned ones.
So for example, for ttl, instead of `long getTtl`, we now use `int getRawTtl`. Gain is using `int` instead of `long`.

Full `raw` method list can be seen [here](
https://github.com/hazelcast/hazelcast/blob/7a290af63624f54450ef05ad2ee695c224d183e7/hazelcast/src/main/java/com/hazelcast/map/impl/record/Record.java#L30).

Reply:
Got it, thanks!